3-UsingAPIs/Week2/homework/ex3-rollAnAce.js

```diff
@@ -10,27 +11,25 @@
 // ! Do not change or remove the next two lines
 const rollDice = require('../../helpers/pokerDiceRoller');
 
-function rollDiceUntil(wantedValue) {
+async function rollDiceUntil(wantedValue) {
   // TODO: rewrite this function using async/await
-  const recurse = () => {
-    return rollDice().then((value) => {
-      if (value !== wantedValue) {
-        return recurse();
-      }
-      return value;
-    });
-  };
-  return recurse();
+  const value = await rollDice()
+  while (value !== wantedValue) {
+    return rollDiceUntil("ACE")
+  }
+  return value
```

Comment:
This is already simpler than the starter code, but it can still be made simpler (easier to understand). Note that the `while` statement functions as an `if` statement because it is never executed more than once because you immediately do a `return` from the while block.

Here is a simpler version:

```js
async function rollDiceUntil(wantedValue) {
  let value = await rollDice();
  while (value !== wantedValue) {
    value = await rollDice();
  }
  return value;
}
```